From: Jiri Olsa <jolsa@redhat.com>
Date: Wed, 18 May 2016 08:16:10 +0200
Subject: perf/x86/intel/uncore: Remove WARN_ON_ONCE in uncore_pci_probe
Git-commit: ef3f00a4d38e01ec0e7ad1b1c8edc2f5667aaa32
Patch-mainline: v4.7-rc1
References: FATE#319705 (supporting patches needed for exit_box)
Signed-off-by: Tony Jones <tonyj@suse.de>

    [suse: 
     adjust for context, no cff9ab2b291e64259d97add48fe073c081afe4e2
     x86/apic: Get rid of apic_version[] array

     adjust for missing 6bcb2db547be8263a98ae9413127df9385b38763
     perf/x86: Move perf_event_intel_uncore.[ch] .. => 
        x86/events/intel/uncore.[ch]
    ]

    perf/x86/intel/uncore: Remove WARN_ON_ONCE in uncore_pci_probe
    
    When booting with nr_cpus=1, uncore_pci_probe tries to init the PCI/uncore
    also for the other packages and fails with warning when they are not found.
    
    The warning is bogus because it's correct to fail here for packages which are
    not initialized. Remove it and return silently.
    
    Fixes: cf6d445f6897 "perf/x86/uncore: Track packages, not per CPU data"
    Signed-off-by: Jiri Olsa <jolsa@kernel.org>
    Cc: stable@vger.kernel.org
    Cc: Peter Zijlstra <peterz@infradead.org>
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

---
 arch/x86/kernel/cpu/perf_event_intel_uncore.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/x86/kernel/cpu/perf_event_intel_uncore.c
+++ b/arch/x86/kernel/cpu/perf_event_intel_uncore.c
@@ -865,7 +865,7 @@ static int uncore_pci_probe(struct pci_d
 		return -ENODEV;
 
 	pkg = topology_phys_to_logical_pkg(phys_id);
-	if (WARN_ON_ONCE(pkg < 0))
+	if (pkg < 0)
 		return -EINVAL;
 
 	if (UNCORE_PCI_DEV_TYPE(id->driver_data) == UNCORE_EXTRA_PCI_DEV) {
